File: assignments/assignment-1-solution/src/CrazyEights/Player/PlayerRegistration.cs
using CrazyEights.Game;

namespace CrazyEights.Player;

public static class PlayerRegistration
{
    public static void Register(GameContext gameContext)
    {
        var defaultName = HumanPlayer.DefaultName;
        Console.Write($"Enter your name (or press Enter for '{defaultName}'):");
        var yourName = Console.ReadLine();
        yourName = string.IsNullOrEmpty(yourName) ? defaultName : yourName.Trim();

        var players = gameContext.Players;
        var hands = HandDealer.Deal(gameContext, 2);
        players.Add(new HumanPlayer(yourName, hands[0]));
        players.Add(new CpuPlayer(hands[1], gameContext.ShowAllHands));
    }
}

File: assignments/assignment-1-solution/src-java/src/main/java/edu/kennesaw/crazy8s/player/PlayerRegistration.java
package edu.kennesaw.crazy8s.player;

import edu.kennesaw.crazy8s.game.GameConsole;
import edu.kennesaw.crazy8s.game.GameContext;

/**
 * Handles registering players for a new game.
 */
public final class PlayerRegistration {
    private PlayerRegistration() {
    }

    public static void register(GameContext gameContext) {
        String defaultName = HumanPlayer.DEFAULT_NAME;
        String prompt = "Enter your name (or press Enter for '" + defaultName + "'): ";
        String yourName = GameConsole.readLineRaw(prompt);
        if (yourName == null || yourName.trim().isEmpty()) {
            yourName = defaultName;
        } else {
            yourName = yourName.trim();
        }

        Players players = gameContext.getPlayers();
        Hand[] hands = HandDealer.deal(gameContext, 2);
        players.add(new HumanPlayer(yourName, hands[0]));
        players.add(new CpuPlayer(hands[1], gameContext.isShowAllHands()));
    }
}
